### PR TITLE
CI: bump up macos-10.15 to macos-11

### PR DESCRIPTION
**Pull request by AkihiroSuda**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macos-10.15, ubuntu-20.04]
+        os: [macos-11, ubuntu-20.04]
```
Comment (jandubois) [inline .github/workflows/test.yml:58]: Any reason not to test on both? Given that macOS is the primary deployment target.

Comment (AkihiroSuda) [reply .github/workflows/test.yml:58]: Updated to support both, but if it makes CI even slower, we should consider removing 10.15.

Comment (jandubois) [reply .github/workflows/test.yml:58]: Will they not run in parallel?

Comment (AkihiroSuda) [reply .github/workflows/test.yml:58]: Seems sometimes yes, sometimes no

Comment (jandubois) [reply .github/workflows/test.yml:58]: There are limits on the number of concurrent jobs per org, but I don't think we should be hitting these limits: https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits

Comment (AkihiroSuda) [reply .github/workflows/test.yml:58]: Our test are still very flaky, and almost impossible to make 4 jobs green. And we can’t restart jobs one by one.

So I guess we should just drop 10.15 CI.

Comment (jandubois) [reply .github/workflows/test.yml:58]: > So I guess we should just drop 10.15 CI.

I guess we have to then; I thought it was mostly the Linux builds failing though, but maybe I'm misremembering.

**Comment by AkihiroSuda** [comment]
Retried several time, but CI can't be greeen.

Seems we have to deflake our tests (and maybe rewrite in Go) before merging this PR...

**Review by jandubois — APPROVED** [review]

**Pull request merged**